calico_containers/adapter/datastore.py

```diff
@@ -590,6 +596,87 @@ def remove_ip_pool(self, version, pool):
             raise KeyError("%s is not a configured IP pool." % pool)
 
     @handle_errors
+    def get_hostnames(self):
+        """
+        Return a list of hostnames
+        :return:  A list of hostnames configured in etcd.
+        """
+        # Get a list of the hosts.
+        hosts = []
+        results = self.etcd_client.read(HOSTS_PATH)
+        for result in results.children:
+            packed = result.key.split("/")
+            # Handle no hosts being configured (etcd returns the parent).
+            if len(packed) != 5:
```

Comment:
Key parsing would be more robust if we compute the number of key elements based on HOSTS_PATH.  
```
key_len = len(HOSTS_PATH.split("/")) + 1
...
    if len(packed) != key_len:
    ...
```